core/src/main/java/com/scalar/db/transaction/consensuscommit/BeforePreparationSnapshotHook.java

```diff
@@ -0,0 +1,10 @@
+package com.scalar.db.transaction.consensuscommit;
+
+public interface BeforePreparationSnapshotHook {
+  BeforePreparationSnapshotHookFuture handle(
```

Comment:
Sorry, one more thing. Why not use the Java `Future` as follows?
```suggestion
  Future<Void> handle(
```

Reply:
I first thought using the custom future would be helpful if we add a new API like `cancel()` to the future. But it also requires some changes in `CommitHandler` to call new APIs. Since we can postpone implementing the custom future until we actually need such new APIs, I'll go ahead and merge your suggestion to simplify things for now.

Reply:
[Done](https://github.com/scalar-labs/scalardb/pull/2271/commits/657558cb56c213c81ce84702dcfbdc5717b0c1aa)